torchgen/gen_vmap_plumbing.py

```diff
@@ -161,13 +168,14 @@ def gen_vmap_inplace_plumbing(native_function: NativeFunction) -> Optional[str]:
 
     unwraps, unwrapped_arg_list = gen_unwraps(schema.arguments.flat_all, cur_level_var)
     bdims_all_none_case = gen_case_where_all_bdims_are_none(sig, schema, cur_level_var)
+    check_escaped = gen_check_escaped("maybe_layer.has_value()", "(inplace)")
```

Comment:
@zou3519, I played around with it, but  I am unsure of how to repro the cases for `gen_vmap_inplace_plumbing` and below in `gen_vmap_plumbing_no_returns` 

I've also seen that there are 24 other instances of the `TORCH_INTERNAL_ASSERT(maybe_layer.has_value());` assert, should I change all of these and add tests?